astroquery/esasky/tests/test_esasky_remote.py

```diff
@@ -1,352 +1,281 @@
             if mission.upper() == "HERSCHEL":
                 assert (isinstance(result[mission.upper()]["1342253595"]["WBS"]["WBS-V_USB_4b"], HDUList))
                 assert (isinstance(result[mission.upper()]["1342253595"]["HRS"]["HRS-H_LSB_4b"], HDUList))
             else:
                 assert (isinstance(result[mission.upper()][0], HDUList))
 
-        result = None
-
-        shutil.rmtree(download_directory)
-
     def test_esasky_query_region_maps(self):
-        result = ESASkyClass.query_region_maps(position="M51", radius="5 arcmin")
+        result = ESASky.query_region_maps(position="M51", radius="5 arcmin")
         assert isinstance(result, TableList)
 
     def test_esasky_query_object_maps(self):
-        result = ESASkyClass.query_object_maps(position="M51")
+        result = ESASky.query_object_maps(position="M51")
         assert isinstance(result, TableList)
 
     @pytest.mark.bigdata
-    def test_esasky_get_images(self):
-        tempdir = tempfile.mkdtemp("ESASkyRemoteTest")
-
-        missions = ESASkyClass.list_maps()
-        # Remove very large map missions & missions with many results
-        # & missions without proper download url (INTEGRAL, SUZAKU, ALMA, AKARI)
+    def test_esasky_get_images(self, tmp_path):
+        missions = ESASky.list_maps()
+        # Remove very large map missions & missions with many results and
+        # missions without proper download url (INTEGRAL, SUZAKU, ALMA, AKARI)
         missions = [mission for mission in missions if mission not in
                     ("HST-OPTICAL", "HST-IR", "HST-UV", "XMM-OM-UV", "INTEGRAL", "SUZAKU", "ALMA", "AKARI")]
 
-        ESASkyClass.get_images(position="M51", missions=missions, download_dir=tempdir)
+        missions = ['XMM', 'Chandra', 'XMM-OM-OPTICAL', 'ISO-IR', 'Herschel', 'JWST_Mid-IR', 'JWST_Near-IR', 'Spitzer']
+        ESASky.get_images(position="M51", missions=missions, download_dir=tmp_path)
 
-        assert (os.path.getsize(tempdir) > 0)
-
-    def test_esasky_get_images_small(self):
-        download_directory = "ESASkyRemoteTest"
-        if not os.path.exists(download_directory):
-            os.makedirs(download_directory)
+        assert os.path.getsize(tmp_path) > 0
 
+    @pytest.mark.skip(reason="Query returns nothing; test checks if tempdir exists")
+    def test_esasky_get_images_small(self, tmp_path):
         # ISO is only ~ 163 kB
         missions = ['ISO-IR']
 
-        ESASkyClass.get_images(position="M6", radius="12arcmin", missions=missions, download_dir=download_directory)
-
+        ESASky.get_images(position="M6", radius="12arcmin", missions=missions,
+                          download_dir=tmp_path)
         for mission in missions:
-            file_path = os.path.join(download_directory, mission)
-            assert file_path[:16] == "ESASkyRemoteTest"
-
-        shutil.rmtree(download_directory)
+            path = tmp_path / mission
+            assert path.parent.exists()
```

Comment:
The updated code is equivalent to
```suggestion
        assert tmp_path.exists()
```
This does not seem like a very interesting test.

Reply:
Indeed, the `test_easasky_get_images_small` test should probably be deleted as the query doesn't actually retrieve anything, nor test that it is retrieving anything.  All it is doing is verifying that pytest's `tmp_path` fixture works.  I'm not sure if this test ever worked.

Reply:
Deleting this test, as it is covered in the test above.